# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

RenderText::PerformLayout() is causing a SIGSEGV for some reasons. 
The error appears to be raised in the Measure method(). This is
causing hap applications to crash, including ACTS test suites.
As a temporary solution we are removing the part of the code that
is raising the segfault. The side effect of this is that texts are 
not properly rendered.

Apply to foundation/ace/ace_engine

Signed-off-by: Francesco Pham <francesco.pham@huawei.com>
Upstream-Status: Inappropriate

diff --git a/frameworks/core/components/text/render_text.cpp b/frameworks/core/components/text/render_text.cpp
index f3135d3c..813d6179 100644
--- a/frameworks/core/components/text/render_text.cpp
+++ b/frameworks/core/components/text/render_text.cpp
@@ -133,8 +133,8 @@ void RenderText::PerformLayout()
     if (pipelineContext) {
         UpdateIfChanged(dipScale_, pipelineContext->GetDipScale());
     }
-    Size size = Measure();
-    SetLayoutSize(GetLayoutParam().Constrain(size));
+    // Size size = Measure();
+    // SetLayoutSize(GetLayoutParam().Constrain(size));
     for (const auto& spanChild : GetChildren()) {
         if (spanChild) {
             const auto& param = GetLayoutParam();
